### ISSUE TITLE
Held effects not removing

### ISSUE THREAD

**Issue by KrlozZo** [GC-spigot/AdvancedEnchantments#3834]
### Describe the bug

When you have a infinite effect with held type enchants, if you remove the item from your inventory with any plugin like "trade or essentials" the effect does not remove and keeps infinite.

### How to reproduce

Take a item that have a held effect enchant then you can use /cllearinventory and you will keep the effect.

### Screenshots / Videos

https://youtu.be/LNAcyAPfAKE

### "/ae plinfo" link

no need

### Server Log

no need

**Comment by SirJben** [comment]
It is happening on 9.15.19 too.

**Comment by GC-spigot** [comment]
Problem with this is, there's no way to detect this properly without doing constant checks of held items, which is bad for performance. We'll add some pro-active checks.

**Comment by ThomasWega** [comment]
We actually do work with most trading plugins just fine. There are already checks for this and now I've also implemented check for /clear command now. The reason this doesn't work is that your specific trade plugin cancels the ```InventoryClickEvent``` for some strange reason. You will need to forward this to them, not us.

However I did take some precautions to support Essentials and other menus better. Just note that your trade plugin won't work and it's not our fault
